src/libraries/System.Private.CoreLib/src/System/Char.cs

```diff
@@ -850,12 +846,9 @@ public static bool IsSymbol(string s, int index)
             }
 
             char c = s[index];
-            if (IsLatin1(c))
-            {
-                return CheckSymbol(GetLatin1UnicodeCategory(c));
-            }
-
-            return CheckSymbol(CharUnicodeInfo.GetUnicodeCategoryInternal(s, index));
+            return CheckSymbol(IsLatin1(c) ?
+                GetLatin1UnicodeCategory(c) :
+                CharUnicodeInfo.GetUnicodeCategoryInternal(s, index));
```

Comment:
Wonder if this is a transform we could have the JIT implicitly do...

Would require some pattern recognition in morph, but seems "feasible" overall?

Reply:
(That said, it's possible this change would actually be a deoptimization if everything gets inlined and `c` is a known constant?)

Reply:
> I could definitely see it being a de-optimization in some cases, but ideally the JIT could have some basic heuristics and constant folding for this to avoid most of the issues.

When I said "this", I was actually referring to my change :)

Reply:
Wouldn't expect this change to impact existing constant folding support and I'd expect it to improve inlinability in general.

A local check shows both get optimized down correctly for `IsPunctuation(' ')`:
```asm
; Method T:B():bool
G_M000_IG01:                ;; offset=0000H
       4883EC28             sub      rsp, 40

G_M000_IG02:                ;; offset=0004H
       B920000000           mov      ecx, 32
       FF15F9792E00         call     [GetLatin1UnicodeCategory(ushort):int]
       8BC8                 mov      ecx, eax
       FF15C1792E00         call     [CheckPunctuation(int):bool]
       90                   nop      

G_M000_IG03:                ;; offset=0018H
       4883C428             add      rsp, 40
       C3                   ret      
; Total bytes of code: 29
```

Reply:
The real method fully optimizes both to `xor eax, eax`

Reply:
> The real method fully optimizes both to xor eax, eax

:smile: